<launch>
    <arg name="map_size_x" default="15.0"/>
    <arg name="map_size_y" default="15.0"/>
    <arg name="map_resolution" default="0.05"/>

    <arg name="start_x" default="0.0"/>
    <arg name="start_y" default="0.0"/>

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find path_searcher)/rviz/display_astar_2d.rviz"/>

    <node pkg="path_searcher" type="run_astar_2d" name="run_astar_2d" output="screen">
        <param name="map/x_size" value="$(arg map_size_x)"/>
        <param name="map/y_size" value="$(arg map_size_y)"/>
        <param name="map/resolution" value="$(arg map_resolution)"/>

        <param name="planning/start_x" value="$(arg start_x)"/>
        <param name="planning/start_y" value="$(arg start_y)"/>
        <param name="planning/allow_diagonal" value="true"/>
    </node>

    <node pkg="map_generator" type="map_generator_2d" name="map_generator_2d" output="screen">
        <param name="init_x" value="0.0"/>
        <param name="init_y" value="0.0"/>

        <param name="map/x_size" value="$(arg map_size_x)"/>
        <param name="map/y_size" value="$(arg map_size_y)"/>

        <param name="Obstacle/lower_w" value="0.3"/>
        <param name="Obstacle/upper_w" value="1.0"/>

        <param name="map/obstacle_number" value="100"/>
        <param name="map/resolution" value="0.05"/>

        <param name="sensor/rate" value="0.5"/>
    </node>
</launch>